drivers/adc-dac/ad74413r/iio_ad74413r.c

```diff
@@ -0,0 +1,763 @@
+{
+	uint32_t n_chan = 0;
+	uint32_t channel_buff_cnt = 0;
+	struct ad74413r_desc *ad74413r_desc = iio_desc->ad74413r_desc;
+	struct ad74413r_channel_config *config = ad74413r_desc->channel_configs;
+	struct iio_channel *chan;
+	struct iio_channel *chan_buffer;
+	struct ad74413r_channel_map channels_info;
+
+	for (uint32_t i = 0; i < AD74413R_N_CHANNELS; i++) {
+		if (!config[i].enabled)
+			continue;
+		channel_buff_cnt += channel_map[config[i].function].num_channels;
+	}
+
+	chan_buffer = calloc(channel_buff_cnt, sizeof(*chan_buffer));
+	if (!chan_buffer)
+		return -ENOMEM;
+
+	iio_desc->iio_dev->channels = chan_buffer;
+
+	/** Add the ADC channels */
+	for (int i = 0; i < AD74413R_N_CHANNELS; i++) {
+		if (!config[i].enabled)
+			continue;
+		channels_info = channel_map[config[i].function];
+		chan = channels_info.channels;
+
+		chan[0].channel = i;
+		chan[0].address = i;
+		chan[0].scan_index = i;
+
+		memcpy(chan_buffer + n_chan, chan, sizeof(*chan));
+		n_chan++;
+		iio_desc->iio_dev->num_ch++;
+	}
+
+	/** Add the DAC channels */
+	for (int i = 0; i < AD74413R_N_CHANNELS; i++) {
+		if (!config[i].enabled)
+			continue;
+		channels_info = channel_map[config[i].function];
+		chan = channels_info.channels;
+		if (channels_info.num_channels == 1)
+			continue;
+
+		chan[1].channel = i;
+		chan[1].address = i;
+
+		memcpy(chan_buffer + n_chan, &chan[1], sizeof(chan[1]));
+		n_chan++;
+		iio_desc->iio_dev->num_ch++;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Enable IIO channels and start the ADC conversions in continuous mode.
+ * @param dev - The iio device structure.
+ * @param mask - Bit mask that specifies the enabled channels.
+ * @return 0 in case of success, error code otherwise
+ */
+static int ad74413r_iio_update_channels(void *dev, uint32_t mask)
+{
+	struct ad74413r_iio_desc *iio_desc = dev;
+
+	iio_desc->active_channels = mask;
+	iio_desc->no_of_active_channels = no_os_hweight8(mask);
+
+	return ad74413r_set_adc_conv_seq(iio_desc->ad74413r_desc, AD74413R_START_CONT);
+}
+
+/**
+ * @brief Stop the ADC conversions.
+ * @param dev - The iio device structure.
+ * @return 0 in case of success, error code otherwise
+ */
+static int ad74413r_iio_buffer_disable(void *dev)
+{
+	struct ad74413r_iio_desc *iio_desc = dev;
+
+	return ad74413r_set_adc_conv_seq(iio_desc->ad74413r_desc,
+					 AD74413R_STOP_PWR_DOWN);
+}
+
+/**
+ * @brief Read a number of samples from each enabled channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to store the samples in.
+ * @param samples - The number of samples
+ * @return The number of samples in case of success, an error code otherwise
+ */
+static int ad74413r_iio_read_samples(void *dev, uint32_t *buf, uint32_t samples)
+{
+	int ret;
+	uint32_t j = 0;
+	uint32_t val;
+	struct ad74413r_iio_desc *iio_desc;
+	struct ad74413r_desc *ad74413r_desc;
+
+	iio_desc = dev;
+	ad74413r_desc = iio_desc->ad74413r_desc;
+
+	for (uint32_t i = 0; i < samples; i++) {
```

Comment:
we should declare the type of the variable at the start of the function (according to the linux drivers code guidelines). Same for `chan_i`. check the entire code for this type of nits